plutus-chain-index/app/Indexer.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE ExplicitForAll #-}
+{-# LANGUAGE GADTs          #-}
+{-# LANGUAGE LambdaCase     #-}
+{-# LANGUAGE TupleSections  #-}
+
+module Main where
+
+import Cardano.Api (Block (Block), BlockHeader (BlockHeader), BlockInMode (BlockInMode), BlockNo (BlockNo), CardanoMode,
+                    ChainPoint (ChainPoint), Hash, NetworkId (Mainnet), SlotNo (SlotNo), chainPointToSlotNo,
+                    deserialiseFromRawBytesHex, proxyToAsType)
+import Cardano.Api qualified as C
+import Cardano.BM.Trace (nullTracer)
+import Cardano.Index.Datum (DatumIndex)
+import Cardano.Index.Datum qualified as Ix
+import Cardano.Protocol.Socket.Client (ChainSyncEvent (Resume, RollBackward, RollForward), runChainSync)
+import Control.Concurrent (threadDelay)
+import Control.Lens.Operators ((^.))
+import Control.Monad (forever, when)
+import Data.IORef (IORef, newIORef, readIORef, writeIORef)
+import Data.List (findIndex)
+import Data.Map (assocs)
+import Data.Maybe (fromJust, fromMaybe)
+import Data.Proxy (Proxy (Proxy))
+import Data.Text (pack)
+import Data.Text.Encoding (encodeUtf8)
+import Index.VSplit qualified as Ix
+import Ledger.TimeSlot (SlotConfig (..))
+import Plutus.ChainIndex.Tx (ChainIndexTx (..))
+import Plutus.Contract.CardanoAPI (fromCardanoTx)
+import Plutus.Script.Utils.V1.Scripts (Datum, DatumHash)
+
+-- We only care about the mainnet
+slotConfig :: SlotConfig
+slotConfig =
+  SlotConfig
+    { scSlotZeroTime = 1596059091000
+    , scSlotLength   = 1000
+    }
+
+networkId :: NetworkId
+networkId = Mainnet
+
+closeToGoguen :: ChainPoint
+closeToGoguen =
+  ChainPoint
+    (SlotNo 39795032)
+    (fromJust $ parseHash "3e6f6450f85962d651654ee66091980b2332166f5505fd10b97b0520c9efac90")
+
+parseHash :: String -> Maybe (Hash BlockHeader)
+parseHash hash =
+  deserialiseFromRawBytesHex (proxyToAsType Proxy) (encodeUtf8 $ pack hash)
```

Comment:
:-/ I had made this simpler in https://github.com/input-output-hk/cardano-node/pull/3619 but we don't have it yet.

Reply:
Yeah. This is just temporary (tm) :)